lib/IRGen/IRGen.cpp

```diff
@@ -484,21 +484,17 @@ performOptimizationsUsingNewPassManger(const IRGenOptions &Opts,
   if (Opts.Sanitizers & SanitizerKind::Address) {
     PB.registerOptimizerLastEPCallback([&](ModulePassManager &MPM,
                                            OptimizationLevel Level) {
-      auto Recover = bool(Opts.SanitizersWithRecoveryInstrumentation &
+      AddressSanitizerOptions Opts;
+      Opts.CompileKernel = false;
+      Opts.Recover = bool(Opts.SanitizersWithRecoveryInstrumentation &
                           SanitizerKind::Address);
-      bool UseAfterScope = false;
-      bool ModuleUseAfterScope = true;
-      bool UseOdrIndicator = Opts.SanitizeAddressUseODRIndicator;
-      llvm::AsanDtorKind DestructorKind = llvm::AsanDtorKind::Global;
-      llvm::AsanDetectStackUseAfterReturnMode UseAfterReturn =
-          llvm::AsanDetectStackUseAfterReturnMode::Runtime;
+      Opts.UseAfterScope = true;
+      Opts.UseAfterReturn = llvm::AsanDetectStackUseAfterReturnMode::Runtime;
       MPM.addPass(
           RequireAnalysisPass<ASanGlobalsMetadataAnalysis, llvm::Module>());
-      MPM.addPass(ModuleAddressSanitizerPass(false, Recover,
-                                             ModuleUseAfterScope,
-                                             UseOdrIndicator, DestructorKind));
-      MPM.addPass(createModuleToFunctionPassAdaptor(AddressSanitizerPass(
-          {false, Recover, UseAfterScope, UseAfterReturn})));
+      MPM.addPass(ModuleAddressSanitizerPass(
+          Opts, /*UseGlobalGC=*/false, Opts.SanitizeAddressUseODRIndicator,
```

Comment:
I think we should set UseGlobalGC to true based on that is what we used to do:

https://github.com/apple/swift/blob/ab4fa0a2046f3d7ba883f6a7d2c9b621d8bedbf3/lib/IRGen/IRGen.cpp#L145